python/pyspark/sql/functions.py

```diff
@@ -2185,6 +2185,12 @@ def udf(f=None, returnType=StringType()):
         duplicate invocations may be eliminated or the function may even be invoked more times than
         it is present in the query.
 
+    .. note:: The user-defined functions do not support conditional execution by using them with
```

Comment:
Hm, should we maybe clarify the output itself is correct if it does not cause the runtime failure by the condition? Maybe I am too much worried but think it might mislead the output is incorrect at all.

Reply:
Yes. I think this is a valid worry. Thanks.